### ISSUE TITLE
Microsoft Phi-3.5 models

### ISSUE THREAD

**Issue by animaldomestico** [ollama/ollama#6449]
- [Phi-3.5-mini-instruct](https://huggingface.co/microsoft/Phi-3.5-mini-instruct)
- [Phi-3.5-MoE-instruct](https://huggingface.co/microsoft/Phi-3.5-MoE-instruct)
- [Phi-3.5-vision-instruct](https://huggingface.co/microsoft/Phi-3.5-vision-instruct)

**Comment by kiradzS** [comment]
I tried, the mini seems to work fine originally, didn't try the others

**Comment by coder543** [comment]
Yes, that one should “just work”. It’s the other two that will likely be more of a problem… especially the vision one, since I don’t think Phi-3 vision *ever* got supported, and now there’s a new version of it.

**Comment by MathiasSchindler** [comment]
Using "ollama run phi3.5" works for a short while and then the output turns completely erratic.

**Comment by NightingaleCen** [comment]
> When I use "ollama run phi3.5", the model replies but it does not stop. It just continues with anything else until the token window is full.

Might be related to ggerganov/llama.cpp#9127 ?

**Comment by terodactilo21** [comment]
Vision model could be awesome.

**Comment by daboe01** [comment]
https://github.com/ggerganov/llama.cpp/pull/9209

**Comment by zcasanova** [comment]
I am also seeing this behaviour

**Comment by smashweaver** [comment]
ollama run phi3.5
pulling manifest 
pulling b5374915da53... 100% ▕██████████████████████████████████████████████████████████████████▏ 2.2 GB                         
pulling c608dc615584... 100% ▕██████████████████████████████████████████████████████████████████▏  149 B                         
pulling c5f9f560a704... 100% ▕██████████████████████████████████████████████████████████████████▏ 1.1 KB                         
pulling 45ac4c662a77... 100% ▕██████████████████████████████████████████████████████████████████▏   60 B                         
pulling 0611867ea67b... 100% ▕██████████████████████████████████████████████████████████████████▏  409 B                         
verifying sha256 digest 
Error: digest mismatch, file must be downloaded again: want sha256:b5374915da534cb93df39f03bd4f2cd5a0c533df0d5e21957dc9556c260be9eb, got sha256:ea6cbaac56d2799594481b797f0e3a2c99a4fef7fba5aab7f0afa7827cb1dce7

**Comment by gaussiangit** [comment]
Is it possible to use [Phi-3.5-vision-instruct](https://huggingface.co/microsoft/Phi-3.5-vision-instruct) ? I believe onnx versions are available but not sure how to port to ollama ?

**Comment by dhiltgen** [comment]
This model architecture is now supported. For example `ollama run phi3.5`
